### PR TITLE
Teacher crud

### PR DESCRIPTION
**Pull request by dhalverson**
## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please describe):

## Problem/feature

Building out CRUD functionality for Teachers model
Rubocop displaying some errors that were not relevant to this Rails application

## Solution

Created:
- Teacher request CRUD spec
- Teachers Controller and CRUD actions, including private teacher params
- Teacher serializer, only current attribute is name
- Rubocop .yml file for modifying warnings

Overall a mirror of the existing Student CRUD with some minor syntax changes for formatting

## Other changes (e.g. bug fixes, UI tweaks, small refactors)
- Rubocop .yml file

## Checklist
- [x] Code compiles correctly
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] All tests passing
- [ ] No rubocop offenses
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by Arique1104** [comment]
Hey @dhalverson and @PhilipDeFraties, thank you so much for building this!  I have a question, where did the Travis CI Build go?

**Comment by PhilipDeFraties** [comment]
> Hey @dhalverson and @PhilipDeFraties, thank you so much for building this! I have a question, where did the Travis CI Build go?

@Arique1104 Looks to me like its there?

**Comment by priyapower** [comment]
@Arique1104 I see the Travis Checks

File: spec/requests/api/v1/teachers_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    expect(teacher[:relationships]).to have_key(:courses)
+    expect(teacher[:relationships][:courses]).to be_a(Hash)
+
+    expect(teacher[:relationships][:courses]).to have_key(:data)
+    expect(teacher[:relationships][:courses][:data]).to be_an(Array)
+  end
+
+  it 'can create a new teacher' do
+    teacher_id = create(:teacher).id
```
Comment (priyapower) [inline spec/requests/api/v1/teachers_request_spec.rb]: if we are testing creation of a user, should we be factory creating a teacher here?

**Review by priyapower — CHANGES_REQUESTED** [review]
I just have a quick question about one of the spec tests

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+AllCops:
+  Exclude:
+    - 'db/**/*'
+    - 'config/**/*'
+    - 'script/**/*'
+    - 'bin/{rails,rake,setup,update}'
+    - !ruby/regexp /old_and_unused\.rb$/
```
Comment (Arique1104) [inline .rubocop.yml:7]: This looks awesome!   Way to make sure files that don't need testing get ignored!

File: app/controllers/api/v1/teachers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module Api
+  module V1
+    class TeachersController < ApplicationController
```
Comment (Arique1104) [inline app/controllers/api/v1/teachers_controller.rb:3]: I thought I saw this with different syntax `Api::V1::TeachersController < ApplicationController` should bring this down to one line.

File: config/routes.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   namespace :api do
     namespace :v1 do
-      resources :students
+      resources :students, :teachers
```
Comment (Arique1104) [inline config/routes.rb:7]: Look at this one liner!  Epic!

File: spec/requests/api/v1/teachers_request_spec.rb (hunk 1 above)
Comment (Arique1104) [reply spec/requests/api/v1/teachers_request_spec.rb]: That's a good point.  I think I remember one of the instructor's advice of making sure not to stub out what you're actually testing.  However, I also think that there is some wiggle room seeing as we are asking our Factory bot to create one for us might be viewed as equivalent to creating one.

**Review by Arique1104 — COMMENTED** [review]
Overall, I'm really pleased with the quality of code here. It was super excited to read and figure out how detailed all of our testing was.  I'm excited to see what our test coverage is because I believe it's at 100% but I'm not a computer program, so I'll leave that to simplecov.  It would be nice to see our test coverage in the commits, along with any rubocop offenses that'll need to get cleaned up later. (NVM, I re-read your commit for the pull request and it was on point.)

I mentioned a one liner but I did not officially 'request the change.'  I just believe that the oneliner is best for readability, but also, if this works and that's how we want to do this, than by all means let's go for it.

Great job team!  Thank you so much for pulling all of this together!  It was truly beautiful!

**Comment by Arique1104** [comment]
> > Hey @dhalverson and @PhilipDeFraties, thank you so much for building this! I have a question, where did the Travis CI Build go?
> 
> @Arique1104 Looks to me like its there?

Hey @PhilipDeFraties, thank you!  I see it now.

File: spec/requests/api/v1/teachers_request_spec.rb (hunk 1 above)
Comment (priyapower) [reply spec/requests/api/v1/teachers_request_spec.rb]: Sorry, let me clarify. I meant, if the test is for "creating a new teacher", if line 65 necessary. I don't see it being used in the test so I was wondering if it was an accident :)

File: app/controllers/api/v1/teachers_controller.rb (hunk 3 above)
Comment (priyapower) [reply app/controllers/api/v1/teachers_controller.rb:3]: This is an update after the Rubocop updates

Comment (PhilipDeFraties) [reply app/controllers/api/v1/teachers_controller.rb:3]: This was an auto-correct from Rubocop

**Review by priyapower — APPROVED** [review]

**Pull request merged**